pyqg/tests/test_twolayer_qg.py

```diff
@@ -1,112 +0,0 @@
-import numpy as np
-import pyqg
-
-def test_the_model(rtol=0.1):
-    """Make sure the results are correct within relative tolerance rtol."""
-
-    year = 360*86400.
-    m = pyqg.QGModel(
-            nx=32,                      # grid resolution
-            ny=None,
-            L=1e6,                      # domain size 
-            W=None,
-            # physical parameters
-            beta=1.5e-11,               # gradient of coriolis parameter
-            rek=5.787e-7,               # linear drag in lower layer
-            rd=30000.0,                 # deformation radius
-            delta=0.25,                 # layer thickness ratio (H1/H2)
-            U1=0.05,                    # upper layer flow
-            U2=0.0,                     # lower layer flow
-            filterfac=18.4,
-            # timestepping parameters
-            dt=12800.,                   # numerical timstep
-            tmax=3*year,          # total time of integration
-            tavestart=1*year,     # start time for averaging
-            taveint=12800.,
-            useAB2=True,
-            # diagnostics parameters
-            diagnostics_list='all'      # which diagnostics to output)
-            )
-
-    # set initial conditions
-    m.set_q1q2(
-            (1e-6*np.cos(2*5*np.pi * m.x / m.L) +
-             1e-7*np.cos(2*5*np.pi * m.y / m.W)),
-            np.zeros_like(m.x) )
-                
-    m.run()
-
-    try:
-        q1 = m.q1 # old syntax
-    except AttributeError:
-        q1 = m.q[0] # new syntax
-        
-    q1norm = (q1**2).sum()
-
-    print 'time:       %g' % m.t
-    assert m.t == 93312000.0
-    
-    ## do we really need this if we have all the other diagnostics?
-    #print 'q1norm:     %.15e' % q1norm
-    #np.testing.assert_allclose(q1norm, 9.723198783759038e-08, rtol)
-    #old value
-    np.testing.assert_allclose(q1norm, 9.561430503712755e-08, rtol)
-    
-    # just skip all the other tests for now
-    return
-    
-    ## raw diagnostics (scalar output)
-    diagnostic_results = {
-        'EKE1': 5.695448642915733e-03,
-        'EKE2': 1.088253274803528e-04,
-        'APEgen': 8.842056320175081e-08,
-        'EKEdiss': 6.368668363708053e-08,        
-    }
-    ## old values
-    #diagnostic_results = {
-    #    'EKE1': 0.008183776317328265,
-    #    'EKE2': 0.00015616609033468579,
-    #    'APEgen': 2.5225558013107688e-07,
-    #    'EKEdiss': 1.4806764171539711e-07,        
-    #}
-    
-    ## need to average these diagnostics
-    avg_diagnostic_results = {
-        'entspec': 5.703438193477885e-07,
-        'APEflux': 9.192940039964286e-05,
-        'KEflux': 1.702621259427053e-04,
-        'APEgenspec': 9.058591846403974e-05,
-        'KE1spec': 3.338261440237941e+03,
-        'KE2spec': 7.043282793801889e+01
-    }
```

Comment:
Sorry, yes I saw that - I should have probably copied the relevant code segments in my comment in the first place. 
So why do these differ. It looks to me like they should be the same (safe for the factor 2*rek).

It looks like I can boil the issue down further. From just the last state of the model I get:

    (np.abs(m.p[-1])**2).mean()
    Out[221]: 1045198.6974801013
    
    (np.abs(m.ph[-1])**2./(m.M**2)).sum()
    Out[222]: 845712.10049875535

Shouldn't these be the same? (Parseval's identity)